### PR TITLE
feat: autocomplete feature for email and password

### LINKED ISSUE #22566: [Feature]: Allow devs to enable/disable autofill in the email and password type inputs
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Devs should be able to disable the autofill in the input widget password and email fields. 
- Add a new field called "Allow autofill" in email and password type inputs and set it to true by default 

### Why should this be worked on?

Devs in financial companies would like to ensure this is a choice as they suggest it might be a safety concern for them. 

Front logo [Front conversations](https://app.frontapp.com/open/top_9lifg)

### PR DESCRIPTION
**Pull request by vsvamsi1**
## Description
This PR brings in the autofill toggle when the user selects either email or password. It is always defaulted to the enabled state, the user need to toggle it off to get the text widget to have autocomplete attribute set to "off".

Fixes #22566 

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?

- Manual
- Cypress

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by vsvamsi1** [comment]
/build-deploy-preview sha=3baa6459f7736d3dc39fb36e033643bac86903a6

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Widgets/Input/Inputv2_spec.js [hunk 1]
```diff
@@ -441,4 +445,40 @@ describe("Input widget V2 - ", () => {
     }
     cy.get(".t--widget-textwidget").should("contain", expected);
   }
+
+  function validateAutocomplteAttribute() {
```
Comment (rajatagrawal) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Widgets/Input/Inputv2_spec.js:449]: typo in the function name : validateAutocompleteAttribute

**Review by jsartisan — APPROVED** [review]

**Comment by vsvamsi1** [comment]
/build-deploy-preview sha=b38000c402c1d5ba5682466b8e33b4b1171b81a1

**Comment by chandannkumar** [comment]
@vsvamsi1 The DP link is not available yet, can you please check

**Review by jsartisan — APPROVED** [review]

**Comment by chandannkumar** [comment]
Can we also add this autocomplete feature for Email & Password field in JSONForm?
cc: @dilippitchika

**Review by jsartisan — APPROVED** [review]

**Comment by vsvamsi1** [comment]
Flaky test case is passing locally.
Screenshot 2023-04-27 at 5 34 34 PM

**Comment by chandannkumar** [comment]
Tested this PR and LGTM
- verified on console if autocomplete is 'off' or not

**Pull request merged**